src/Android/MainActivity.cs

```diff
@@ -239,18 +239,22 @@ protected override void OnActivityResult(int requestCode, Result resultCode, Int
                 string fileName = null;
                 if (data != null && data.Data != null)
                 {
-                    uri = data.Data;
-                    fileName = AndroidHelpers.GetFileName(ApplicationContext, uri);
+                    if (!data.Data.ToString()?.Contains(Constants.PACKAGE_NAME) ?? false)
```

Comment:
🤔 Can `ToString()` return `null` when `data.Data` is not `null` (given the if check above)? 
🤔 If so, do you think the next would be easier to read?
```suggestion
                    if (data.Data.ToString()?.Contains(Constants.PACKAGE_NAME) != true)
```

Reply:
Rider is telling me `Uri.ToString()` can return null so I'm trusting it.  Also agree your modification is easier to read 👍